### ISSUE TITLE
Merlin doesn't correctly resolve modules overriding stdlib ones.

### ISSUE THREAD

**Issue by ejgallego** [ocaml/merlin#1003]
In OCaml 4.07/4.08, if I have a file:
```ocaml
let () =
  let s = List.foo in
  print_endline s
```
where `list.ml` is also in my project, and exports `foo`, my project will build correctly [under dune], however merlin insists to resolve the `List` module to the one in `Stdlib`, thus I get:
```
Unbound value List.foo
```

This is creating some pain with existing projects and OCaml 4.08 due to the new modules in that release.

**Comment by trefis** [comment]
That's a known bug with our 4.08 backend yes.
And we're planning to resolve that "soon".

I'm a bit surprise that there is a problem on 4.07 though, that is: I believe on 4.07 the compiler behaves as merlin.

**Comment by ejgallego** [comment]
Hi @trefis , thanks for looking into this. I can reproduce with 4.07.1 .

**Comment by trefis** [comment]
Hm. Interesting.
Then it's not the bug I expected, and there must be dune shenanigans in the mix.

I'll investigate.

**Comment by ejgallego** [comment]
Solved in merlin 3.3.3, thanks a lot!
